.github/workflows/ci.yml

```diff
@@ -0,0 +1,175 @@
+
+name: Build and Upload Artifacts
+
+on:
+  push:
+    branches:
+      - "develop"
+      - "main"
+
+    # NOTE:
+    # "on tags" is used to trigger the release process instead of
+    # "on release" due to current release process requirements.
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+
+  pull_request:
+    branches:
+      - "develop"
+      - "main"
+
+  # Allow manually triggering of the workflow.
+  workflow_dispatch:
+    is_release:
+      description: "Indicates whether to treat this run as a release. This input is only intended for testing purposes. Releases are to be performed on tag creation."
+      type: string
+      required: true
+      default: "false"
+
+env:
+  TMP_SOURCES_DIR: ${{ github.workspace }}/tmp/src
+  TMP_PDF_DOCS_ARTIFACT_NAME: docs.tmp
+  RELEASE_SOURCES_DIR: ${{ github.workspace }}/release/src
+  RELEASE_PDF_DOCS_DIR: ${{ github.workspace }}/release/docs/pdf
+
+jobs:
+  event_configuration:
+    name: Determine event configuration
+    runs-on: ubuntu-latest
+    outputs:
+      is_release: ${{ steps.run_type.outputs.IS_RELEASE }}
+      example_apps_artifact_name: ${{ steps.artifact_names.outputs.EXAMPLES_ARTIFACT_NAME }}
+      src_artifact_name: ${{ steps.artifact_names.outputs.SOURCES_ARTIFACT_NAME }}
+      html_docs_artifact_name: ${{ steps.artifact_names.outputs.HTML_DOCS_ARTIFACT_NAME }}
+      pdf_docs_artifact_name: ${{ steps.artifact_names.outputs.PDF_DOCS_ARTIFACT_NAME }}
+      pdf_docs_artifact_name_release: ${{ steps.artifact_names.outputs.PDF_DOCS_ARTIFACT_NAME_RELEASE }}
+    steps:
+      - name: Determine workflow run type
+        id: run_type
+        run: |
+          IS_RELEASE=false
+          IS_TAGGED=false
+          if [ "${{ startsWith(github.ref, 'refs/tags/v') }}" = "true" ]; then
+            IS_RELEASE=true
+            IS_TAGGED=true
+          fi
+          if [ "${{ inputs.is_release }}" = "true" ]; then
+            IS_RELEASE=true
+          fi
+          echo "IS_RELEASE=$IS_RELEASE" >> $GITHUB_OUTPUT
+          echo "IS_RELEASE=$IS_RELEASE" >> $GITHUB_ENV
+          echo "IS_TAGGED=$IS_TAGGED" >> $GITHUB_ENV
+
+      - name: Determine artifact names
+        id: artifact_names
+        run: |
+          if [ "$IS_RELEASE" = "true" ]; then
+            if [ "$IS_TAGGED" = "true" ]; then
+              version=$(echo "${{ github.ref_name }}" | sed "s/\./_/g")
+            else
+              version="test"
+            fi
+            echo "EXAMPLES_ARTIFACT_NAME=xcore_voice_examples_$version" >> $GITHUB_OUTPUT
+            echo "SOURCES_ARTIFACT_NAME=xcore_voice_sources_$version" >> $GITHUB_OUTPUT
+            echo "HTML_DOCS_ARTIFACT_NAME=xcore_voice_html_docs_$version" >> $GITHUB_OUTPUT
+            echo "PDF_DOCS_ARTIFACT_NAME=$TMP_PDF_DOCS_ARTIFACT_NAME" >> $GITHUB_OUTPUT
+            echo "PDF_DOCS_ARTIFACT_NAME_RELEASE=xcore_voice_pdf_docs_$version" >> $GITHUB_OUTPUT
+          else
+            echo "EXAMPLES_ARTIFACT_NAME=xcore_voice_examples" >> $GITHUB_OUTPUT

```

Comment:
I think this line here might cause a problem with the Jenkins file. It looks like Jenkins might expect the artifact to be `sln_voice_main_example_apps`. Any opinion on the artifact names set on these lines (both in the `release` and `non-release` modes?